### PR TITLE
Enable builds on OpenCloudOS

### PR DESCRIPTION
**Pull request by xiacunshun**
https://github.com/intel/eth-fast-fabric/issues/7

### CODE REVIEW

File: CommonInstall/util_init.pl [hunk 1]
```diff
@@ -324,8 +327,10 @@ ()
 		$CUR_DISTRO_VENDOR = "redhat";
 	} elsif ( -s "/etc/almalinux-release" ) {
 		$CUR_DISTRO_VENDOR = "redhat";
-	} elsif ( -s "/etc/circle-releas" ) {
+	} elsif ( -s "/etc/circle-release" ) {
 		$CUR_DISTRO_VENDOR = "redhat";
+•       } elsif ( -s "/etc/opencloudos-release" ) {
```
Comment (acgoldma) [inline CommonInstall/util_init.pl:332]:
```
•
```
Looks like copy paste error.

File: CommonInstall/util_init.pl [hunk 2]
```diff
@@ -346,6 +351,7 @@ ()
 			"rocky" => "redhat",
 			"almalinux" => "redhat",
 			"circle" => "redhat",
+			"ocs" => "opencloudos",
```
Comment (acgoldma) [inline CommonInstall/util_init.pl:354]: for redhat-like OS we should use redhat for the vendor, just like centos and rocky do.

See https://github.com/intel/eth-fast-fabric/pull/1/commits/97d8054919c8137a3d56167e011c1c05e5b4c9f3

**Review by acgoldma — CHANGES_REQUESTED** [review]

File: CommonInstall/util_init.pl (hunk 1 above)
Comment (xiacunshun) [reply CommonInstall/util_init.pl]: fixed.

File: CommonInstall/util_init.pl (hunk 2 above)
Comment (xiacunshun) [reply CommonInstall/util_init.pl]: We have something in common, but we are not based on redhat and are not redhat-like exactly.

File: MakeTools/funcs-ext.sh [hunk 3]
```diff
@@ -825,6 +831,9 @@ function os_vendor_version()
 	mandrake)
 		rval=`cat /etc/mandrake-release | cut -d' ' -f4`
 		;;
+	opencloudos)
+		rval=`cat /etc/opencloudos-release | cut -d' ' -f4`
```
Comment (acgoldma) [inline MakeTools/funcs-ext.sh:835]: I downloaded the 8.6 rpm and `/etc/opencloudos-release` appears to only have 2 fields. so I think this should be `| cut -d' ' -f2`

File: get_id_and_versionid.sh [hunk 4]
```diff
@@ -101,6 +107,8 @@ else
 		v2=$(grep PATCHLEVEL /etc/SuSE-release | cut -d' ' -f3)
 		rval=${v1}.$v2
 		;;
+	ocs)
+		rval=`cat /etc/opencloudos-release | cut -d' ' -f4`
```
Comment (acgoldma) [inline get_id_and_versionid.sh:111]: need the break statement `;;`

Also should be `-f2` here as well.

**Review by acgoldma — COMMENTED** [review]
We will test the patch does not break support for our currently supported OSes, but we do not currently plan to support OpenCloudOS, so you will need to verify by following the README to build and test it if you want this included in a future release.

File: MakeTools/funcs-ext.sh (hunk 3 above)
Comment (xiacunshun) [reply MakeTools/funcs-ext.sh:835]: I forgot to mention that we are working on a stream release which is different from 8.6.
https://mirrors.opencloudos.tech/opencloudos-stream/updates/2301/x86_64/Packages/opencloudos-stream-release-2301-6.ocs1.noarch.rpm

cat /etc/opencloudos-release
OpenCloudOS Stream release 2301

Comment (xiacunshun) [reply MakeTools/funcs-ext.sh:835]: opencloudos 8.6 is based on centos, but the stream release is not.

File: get_id_and_versionid.sh (hunk 4 above)
Comment (xiacunshun) [reply get_id_and_versionid.sh:111]: fixed.

**Comment by xiacunshun** [comment]
> We will test the patch does not break support for our currently supported OSes, but we do not currently plan to support OpenCloudOS, so you will need to verify by following the README to build and test it if you want this included in a future release.

ok. I saw that Circle linux has been supported, so I try to follow that.
Nonetheless, we still hope opencloudos can be supported. thx.

File: CommonInstall/util_init.pl [hunk 5]
```diff
@@ -239,6 +239,9 @@ ($)
 	} elsif ($vendor eq "fedora") {
 		$rval=`cat /etc/fedora-release | cut -d' ' -f4`;
 		chop($rval);
+	} elsif ($vendor eq "opencloudos") {
```
Comment (acgoldma) [inline CommonInstall/util_init.pl:242]: it should not reach this as we will enter block above at line 206.
So this is not needed.

File: CommonInstall/util_init.pl [hunk 6]
```diff
@@ -324,8 +327,10 @@ ()
 		$CUR_DISTRO_VENDOR = "redhat";
 	} elsif ( -s "/etc/almalinux-release" ) {
 		$CUR_DISTRO_VENDOR = "redhat";
-	} elsif ( -s "/etc/circle-releas" ) {
+	} elsif ( -s "/etc/circle-release" ) {
 		$CUR_DISTRO_VENDOR = "redhat";
+	} elsif ( -s "/etc/opencloudos-release" ) {
```
Comment (acgoldma) [inline CommonInstall/util_init.pl:332]: If you want to enable for the stream release, then I think you should check for the `/etc/opencloudos-stream-release` file.

File: MakeTools/funcs-ext.sh (hunk 3 above)
Comment (acgoldma) [reply MakeTools/funcs-ext.sh:835]: this file is a symlink to the stream file. I think you should read from the stream file `opencloudos-stream-release` as the formats seem different.

File: get_id_and_versionid.sh (hunk 4 above)
Comment (acgoldma) [reply get_id_and_versionid.sh:111]: Same here, should read from `opencloudos-stream-release`

**Review by acgoldma — COMMENTED** [review]
Okay looks like the stream version does not include the `/etc/redhat-release`, so using redhat vendor is not correct.

Minor chnage to read from `opencloudos-stream-release` file and also looks like you may need to update `update_eth_spec.sh` also.

**Comment by ToddRimmer** [comment]
please email todd.rimmer@intel.com with a little more about your environment and opportunity.  What NICs are you using?  What workloads?  Glad to discuss more, If we learn some more, we could explore making openCloudOS a more officially supported distro for IEFS.

File: MakeTools/funcs-ext.sh (hunk 3 above)
Comment (xiacunshun) [reply MakeTools/funcs-ext.sh:835]: ok, fixed it.

File: CommonInstall/util_init.pl (hunk 5 above)
Comment (xiacunshun) [reply CommonInstall/util_init.pl:242]: fixed.

**Comment by acgoldma** [comment]
@xiacunshun , Thank you for all you work. We will evaluate this patch and if there are no issues, we will merge this with a future release.

**Review by acgoldma — APPROVED** [review]
Regression Testing Passed.

**Pull request merged**